apps/ui-library/content/docs/infinite-query-hook.mdx

```diff
@@ -0,0 +1,225 @@
+| --------------- | --------------------------------------------------------- | ---------------------------------------------------------------- |
+| `tableName`     | `string`                                                  | **Required.** The name of the Supabase table to fetch data from. |
+| `columns`       | `string`                                                  | Columns to select from the table. Defaults to `'*'`.             |
+| `pageSize`      | `number`                                                  | Number of items to fetch per page. Defaults to `20`.             |
+| `trailingQuery` | `(query: SupabaseSelectBuilder) => SupabaseSelectBuilder` | Function to apply filters or sorting to the Supabase query.      |
+
+## Return type
+
+data, count, isSuccess, isLoading, isFetching, error, hasMore, fetchNextPage
+
+| Prop            | Type          | Description                                                                         |
+| --------------- | ------------- | ----------------------------------------------------------------------------------- |
+| `data`          | `TableData[]` | An array of fetched items.                                                          |
+| `count`         | `number`      | Number of total items in the database. It takes `trailingQuery` into consideration. |
+| `isSuccess`     | `boolean`     | It's true if the last API call succeeded.                                           |
+| `isLoading`     | `boolean`     | It's true only for the initial fetch.                                               |
+| `isFetching`    | `boolean`     | It's true for the initial and all incremental fetches.                              |
+| `error`         | `any`         | The error from the last fetch.                                                      |
+| `hasMore`       | `boolean`     | Whether the query has finished fetching all items from the database                 |
+| `fetchNextPage` | `() => void`  | Sends a new request for the next items                                              |
+
+## Type-safety
+
+The hook will use the typed defined on your Supabase client if they're setup ([more info](https://supabase.com/docs/reference/javascript/typescript-support)).
+
+The hook also supports an custom defined result type by using `useInfiniteQuery<T>`. For example, if you have a custom type for `Product`, you can use it like this `useInfiniteQuery<Product>`.
+
+## Examples
+
+### With sorting
+
+```tsx
+const { data, fetchNextPage } = useInfiniteQuery({
+  tableName: 'products',
+  columns: '*',
+  pageSize: 10,
+  trailingQuery: (query) => query.order('created_at', { ascending: false }),
+})
+
+return (
+  <div>
+    {data.map((item) => (
+      <ProductCard key={item.id} product={item} />
+    ))}
+    <Button onClick={fetchNextPage}>Load more products</Button>
+  </div>
+)
+```
+
+### With filtering on search params
```

Comment:
this need a tiny subheading with something like this?

"Add a search param to your url like site.com/?=hello"